test/integration.ts

```diff
@@ -0,0 +1,236 @@
+import StreamRegistryJson from "../lib/streamregistry/StreamRegistryV3.json"
+import { getAddress, parseEther } from "ethers/utils"
+
+import type { StreamRegistryV3 } from "../lib/types/StreamRegistryV3"
+
+import CoreAPIClient from "../src/CoreAPIClient"
+
+const { log } = console
+
+import { Chains } from "@streamr/config"
+import assert from "assert"
+
+const {
+    ethereum: {
+        id: networkId,
+        rpcEndpoints: [{
+            url: ETHEREUM_SERVER_URL,
+        }],
+        contracts: {
+            "DATA-token": dataTokenAddress,
+            "Marketplace": MARKETPLACE_ADDRESS,
+        }
+    },
+    streamr: {
+        rpcEndpoints: [{
+            url: MATIC_SERVER_URL,
+        }],
+        contracts: {
+            "StreamRegistry": STREAM_REGISTRY_ADDRESS,
+        }
+    }
+} = Chains.load("development")
+
+const DEVOPS_KEY = "0x628acb12df34bb30a0b2f95ec2e6a743b386c5d4f63aa9f338bec6f613160e78" // 0xa12Ccb60CaD03Ce838aC22EaF2Ce9850736F154f
+const adminKey = "0x5e98cce00cff5dea6b454889f359a4ec06b9fa6b88e9d69b86de8e1c81887da0" // 0xa3d1f77acff0060f7213d7bf3c7fec78df847de1
+const prefundedKey = "0xe5af7834455b7239881b85be89d905d6881dcb4751063897f12be1b0dd546bdb"
+// const productId = "0xf3f975e1837042d98bb50bf0638789dbf58bb7efc3e84253900" + Date.now()
+// const productId = "0x0000000000000000000000000000000000000000000000000000000000000001"
+
+const STREAMR_API_URL = "http://localhost/api/v2"
+
+const watcherEnv: NodeJS.ProcessEnv = {
+    STREAM_REGISTRY_ADDRESS,
+    MARKETPLACE_ADDRESS,
+    NETWORK_ID: networkId.toString(),
+    ETHEREUM_SERVER_URL,
+    MATIC_SERVER_URL,
+    STREAMR_API_URL,
+    DEVOPS_KEY,
+    LAST_BLOCK_DIR: ".",
+}
+const watcherExecutable = "dist/src/main.js"
+
+const provider = new JsonRpcProvider(ETHEREUM_SERVER_URL)
+const watcherWallet = new Wallet(DEVOPS_KEY, provider)
+const wallet = new Wallet(adminKey, provider)
+
+const sidechainProvider = new JsonRpcProvider(MATIC_SERVER_URL)
+const sidechainWallet = new Wallet(adminKey, sidechainProvider)
+
+const tokenAddress = getAddress(dataTokenAddress)
+const marketAddress = getAddress(MARKETPLACE_ADDRESS)
+const registryAddress = getAddress(STREAM_REGISTRY_ADDRESS)
+
+const token = new Contract(tokenAddress, TokenJson.abi, wallet)
+const market = new Contract(marketAddress, MarketplaceJson.abi, wallet)
+const registry = new Contract(registryAddress, StreamRegistryJson.abi, sidechainWallet) as unknown as StreamRegistryV3
+
+const streamIdPath = "/test" + Date.now()
+const streamId = wallet.address.toLowerCase() + streamIdPath
+
+const ownerApiClient = new CoreAPIClient(
+    STREAMR_API_URL,
+    CoreAPIClient.DEFAULT_FETCH_FUNC,
+    CoreAPIClient.DEFAULT_GET_SESSION_TOKEN_FUNC,
+    adminKey
+)
+const trustedApiClient = new CoreAPIClient(
+    STREAMR_API_URL,
+    CoreAPIClient.DEFAULT_FETCH_FUNC,
+    CoreAPIClient.DEFAULT_GET_SESSION_TOKEN_FUNC,
+    DEVOPS_KEY
+)
```

Comment:
I'd move these variables closer to where they are used and I'd narrow the scope. It's a bit unnecessary to scroll up and down to see where variables are declared.

Reply:
moving them as close as possible would be a bit... misleading in the sense that there is currently only one test. Stuff might need to be moved out after adding another test. I think more generally the way our tests are laid our and formatted could be a wider discussion. I wouldn't change this now.

Reply:
If it is used only once then the rationale is even heavier. Are you absolutely sure there is going to be another test that uses all of those variables?

Reply:
it's more like: if there are many test cases, will all of them use these global variables? Where should we put them? So far I've put them outside the describe. They could be inside. Should we even require a describe? Just `it` works too. I don't think simple "put it close to the usage" is the best here, if the result is that every test looks different.

Reply:
That's just speculation. Will there ever be another test? Write code as the current situation is/requires and don't speculate about unknown future, right?

Reply:
There will be other tests in other files, and it will be good for readability if they're all formatted in the same way. This is my formatting for now, until further general discussion on how tests should be laid out.